### PR TITLE
Prevent default event when input is empty and key pressed is `COMMA`

### PR DESCRIPTION
**Pull request by adamgruber**
Typing a `COMMA` in a taggle input is expected to call the submit method and attempt to create a new tag, unless the input is empty. In this case it is possible for a user to inadvertently create a tag with leading spaces.

**Example**

User types something like this:
```
,   my tag
```
which results in a new tag being added with leading spaces:
```
   my tag
```

This PR fixes this edge case by preventing a `COMMA` from being typed in when the input is empty.

### CODE REVIEW

**Comment by okcoker** [comment]
@adamgruber good catch. Can you add the appropriate tests for this case?

**Comment by okcoker** [comment]
Just had a thought. We probably also want to make sure comma is part of `submitKeys` before preventing default as well in case someone allows comma's in their tags.

**Comment by adamgruber** [comment]
I messed around with adding a test but wasn't exactly sure what to assert on. Thought about testing that the input value is not `,` but the `simulateKeyboardEvent` method doesn't actually update the input value.

**Comment by adamgruber** [comment]
@okcoker Sorry to bug, but any chance of getting some feedback on this or even a merge? I can explore an alternate solution that may be more testable if that helps.

File: src/taggle.js [hunk 1]
```diff
@@ -727,6 +727,11 @@
             this.pasting = true;
         }
 
+        if (this._isConfirmKey(key) && key === COMMA && this.input.value === '') {
```
Comment (okcoker) [inline src/taggle.js]: Why is there a check for `key === COMMA` when `this._isConfirmKey(key)` should handle that case? If a user doesn't have `COMMA` as a confirm key, it should be allowed so we can probably remove `key === COMMA`. Thoughts?

**Comment by okcoker** [comment]
Hey @adamgruber just saw the forced pushed. I would like a way to have a test for this before getting merged in but haven't had time lately.

File: src/taggle.js (hunk 1 above)
Comment (adamgruber) [reply src/taggle.js]: `isConfirmKey` could also match `TAB` or `ENTER` in which case we don't want to prevent default.

**Comment by adamgruber** [comment]
@okcoker I opened up #141 as an alternate solution to this issue. It's more straightforward and testable 😃 . Feel free to close this if that seems like a better approach.
